### ISSUE TITLE
Facebook plugin

### ISSUE THREAD

**Issue by mitar** [eloquence/freeyourstuff.cc#36]
I am not sure if this project is for liberating only public content or also private content.

I have tried few times in the past to create a way to download all my Facebook content. I want more than what I can get through their download archive approach:
* I would like to have also all my likes and comments I made
* I would like to have all comments and likes others made on my content

I have tried to use Facebook API to get my own data, but it seems it is pretty limited:
* it does not allow anymore access to messages
* it does not allow to list your own friends

The best approach I have found is to scrape from user's "activity log" all activity you made, and then you can get post IDs and also fetch the rest through Facebook API, if necessary. I think having a browser extension which scrapes "activity log" and infinity loads it until the end. We can probably go around rate limits by going slowly the first time and then remembering where we left the last time.

**Comment by eloquence** [comment]
Hi Mitar,

So far it's only public stuff, but there's no principled reason for that. If we add private stuff, we should make sure we have some notion of the public/private distinction in data and the UI, so users don't accidentally publish private data through the extension's one-click publish feature.

With that said, I have some concerns about importing _other_ people's comments on Facebook etc., even if they're on your own posts. Generally, I try to maintain a line of "This extension only lets you download things you authored/created". Perhaps that would be a good place to start with a Facebook plugin, and then we can think more about how we address issues of consent/copyright/ToS for content the user didn't author...

**Comment by mitar** [comment]
For me it is really about archiving/backuping my digital trail. If my digital trail spawned comments and likes, I want to preserve that as well. I do not really care about legality of this, in my mind it is something like that: people wrote comments on my posts, presumably because they wanted from me to read them, so if I save all of them to my computer so that they are preserved, they probably agree with that as well.

But I do understand that this view might not be shared by everyone.

**Comment by mitar** [comment]
cc @antoinemcgrath

**Comment by eloquence** [comment]
That makes sense, mitar. I think the use case is definitely ethically defensible, but given that people may already make ToS arguments against the extension, I want to play it safe with other people's contributions unless we get the explicit okay from Facebook for this kind of data extraction for personal use. I unfortunately don't have a good contact -- perhaps someone who's involved with their data export team would be ideal.

**Comment by mitar** [comment]
So if you feel uncomfortable with this, maybe then this should be an independent extension?

**Comment by mitar** [comment]
Or maybe we could have something like Greasemonkey where people could add their own external plugins. So then this could be an external plugin.

**Comment by eloquence** [comment]
Yeah, an external/loadable plugin would work just fine, though we'd still need to do the work in the core extension to support private vs. public data.

**Comment by napsternxg** [comment]
After following the discussion and the points raised by @mitar and @eloquence I think twitter might be a safer option to include as a plugin. As the tweets are all personal data and they don't include data from other users - thereby not violating privacy. 

I have recently started a google form https://goo.gl/forms/3fup3Y7TN9seHF2e2 where I plan to collect twitter handles of users who want to make their tweet data available for personal use (just like freeyourstuff.cc makes). I was looking at ways to easy the process when I found out about this project. 

Not only is personal archiving of social media data useful for individuals to keep track of their online activity, but it also allows them the freedom to contribute their data to research initiatives. Most of the social interactions are happening on social media sites these days, however, most of these sites prohibit redistribution of the data extracted from their sites. This creates a bias in terms of who gets to do social research on social media data, accuracy of the research results, and prior-approval by the social media site before publishing such results (if the site is a sponsor of the research).
